### PR TITLE
feat(dialog): allow extra content in web

### PR DESCRIPTION
**Pull request by sergivillar**

### CODE REVIEW

File: src/dialog.tsx [hunk 1]
```diff
@@ -168,6 +185,7 @@ const Dialog: React.FC<DialogProps> = (props) => {
                     {message}
                 </Text3>
             </div>
+            {extra && <Box paddingTop={8}>{extra}</Box>}
```
Comment (sergivillar) [inline src/dialog.tsx:188]: this is the only UI change

File: src/dialog.tsx [hunk 2]
```diff
@@ -257,6 +275,34 @@ const useNativeDialog = ({
     }, [onAcceptRef, onCancelRef, acceptText, cancelText, showCancel, message, title, renderNative]);
 };
 
+type NativeModalDialogProps = ModalDialogProps & {
+    dialogAcceptButton: string;
+    dialogCancelButton: string;
+};
+
+const NativeModalDialog = (props: NativeModalDialogProps) => {
```
Comment (sergivillar) [inline src/dialog.tsx:283]: this is just a Wrapper to the native behaviour

File: playroom/snippets.tsx [hunk 3]
```diff
@@ -1367,9 +1367,29 @@ const alertSnippets = [
   }
 >
   Open two buttons destructive
-</ButtonPrimary>  
+</ButtonPrimary>
       `,
     },
```
Comment (yceballost) [inline playroom/snippets.tsx:1372]: snippet doesn't works in playroom
https://ibit.ly/bQN1

**Review by yceballost — CHANGES_REQUESTED** [review]

File: src/__stories__/dialog-story.tsx [hunk 4]
```diff
@@ -46,6 +47,21 @@ export const Default: StoryComponent = () => (
                 Open two buttons destructive
             </ButtonPrimary>
         </ButtonLayout>
+        <ButtonLayout>
+            <ButtonPrimary
+                onPress={() =>
```
Comment (yceballost) [inline src/__stories__/dialog-story.tsx:52]: This example should not have cancel button

**Review by yceballost — CHANGES_REQUESTED** [review]

**Review by yceballost — APPROVED** [review]
🤟

File: src/dialog.tsx [hunk 5]
```diff
@@ -513,22 +554,27 @@ export default class DialogRoot extends React.Component<DialogRootProps, DialogR
 }
 
 const showDialog =
-    (showCancel = false) =>
+    (showCancel = false, forceNonNative: boolean) =>
```
Comment (pladaria) [inline src/dialog.tsx:557]: I prefer an object:
```js
const showDialog = 
    ({showCancel = false, forceNonNative = false}: {showCancel: boolean, forceNonNative: boolean}) => ...
```

This way instead of:
```js
showDialog(true, false)
```
you would read:
```js
showDialog({showCancel: true, forceNonNative: false})
```

File: src/dialog.tsx [hunk 6]
```diff
@@ -272,7 +322,7 @@ const ModalDialog = (props: ModalDialogProps) => {
             `To use @telefonica/mistica components you must instantiate <ThemeContextProvider> as their parent.`
         );
     }
-    const renderNative = isWebViewBridgeAvailable();
+    const renderNative = !props.forceNonNative && isWebViewBridgeAvailable();
```
Comment (pladaria) [inline src/dialog.tsx:325]: double negation, perhaps you could rename `forceNonNative` to `forceWeb`

**Review by pladaria — APPROVED** [review]

File: src/dialog.tsx [hunk 7]
```diff
@@ -257,6 +279,34 @@ const useNativeDialog = ({
     }, [onAcceptRef, onCancelRef, acceptText, cancelText, showCancel, message, title, renderNative]);
 };
 
+type NativeModalDialogProps = ModalDialogProps & {
+    dialogAcceptButton: string;
+    dialogCancelButton: string;
+};
+
+const NativeModalDialog = (props: NativeModalDialogProps) => {
+    const renderNative = isWebViewBridgeAvailable();
```
Comment (atabel) [inline src/dialog.tsx:288]: why is this condition repeated now?

File: src/dialog.tsx [hunk 8]
```diff
@@ -312,34 +362,19 @@ const ModalDialog = (props: ModalDialogProps) => {
...
         if (!animationsSupported(platformOverrides)) {
             addKeyDownListener();
         }
 
-        if (
-            (renderNative || !animationsSupported(platformOverrides)) &&
-            props.isClosing &&
-            props.onCloseTransitionEnd
-        ) {
+        if (!animationsSupported(platformOverrides) && props.isClosing && props.onCloseTransitionEnd) {
```
Comment (atabel) [inline src/dialog.tsx:370]: why this change?

File: src/dialog.tsx (hunk 7 above)
Comment (sergivillar) [reply src/dialog.tsx:288]: true, not needed here, because will be always `true`, removed

File: src/dialog.tsx (hunk 8 above)
Comment (sergivillar) [reply src/dialog.tsx:370]: just move the `renderNative` to new component, but as renderNative is not needed there, this could be restored as it was too

**Review by atabel — APPROVED** [review]

**Comment by atabel** [comment]
I miss an screenshot for the new case

**Review by atabel — APPROVED** [review]
